### PR TITLE
Added Phylomizer runner

### PR DESCRIPTION
**Pull request by qiyunzhu**
Here I added Bash code to fire [Phylomizer](https://github.com/Gabaldonlab/phylomizer), the gene family tree building pipeline developed by Toni's team. @scapella the main developer of Phylomizer has been helping us with the setup.Phylomizer is quite flexible. I adopted the [KMM protocol](https://github.com/Gabaldonlab/phylomizer/blob/master/config/PhylomeDB%20Configuration%20-%20KMM) and made a few modifications. In brief, the procedures are:

1. Start with a multi-Fasta file containing protein sequences of a gene family.
2. Run three multiple sequence aligners independently: KAlign, MUSCLE and MAFFT.
3. Extract a best consensus alignment, using M-Coffee.
4. Perform quality trimming of the alignment, using TrimAl.
5. Perform maximum likelihood tree reconstruction, using PhyML, with a variety of common amino acid substitution models: JTT, WAG, MtREV, VT, LG, Blosum62 and Dayhoff
6. Rank the resulting trees by log likelihood, and retain the best tree.
7. This tree has branch lengths and nodal supports (bootstrap values).

Note: alignments with less than 10 sequences will be considered useless and discarded.

This protocol can and should be discussed on the meeting, in order to achieve our best needs. Next, I will try to connect OrthoFinder with Phylomizer.

@ekopylova @wasade @sjanssen2 @antgonza @tanaes

### CODE REVIEW

File: benchmark/run_phylomizer.sh [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+$verbose && echo "Running Phylomizer .."
+
+# set up
+PWD=$(pwd)
+mkdir -p ${working_dir}/phylomizer
+if [ "${phylomizer_config_fp}" == "None" ]
+then
+    phylomizer_config_fp=${working_dir}/phylomizer/config.txt
+    # modified KMM protocol
+    # minimum number of sequences: 10
```
Comment (ekopylova) [inline benchmark/run_phylomizer.sh:40]: is 10 the suggested default?

File: benchmark/run_phylomizer.sh [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    # copy the best tree to the gene tree directory
+    cp input.tree.phyml.ml.${best_model}.nw ${gene_tree_dir}/$gene.nwk
+
+    user_time=$(echo $TIME | awk '{print $1;}')
+    wall_time=$(echo $TIME | awk '{print $2;}')
+    total_user_time=$(echo $total_user_time + $user_time | bc)
+    total_wall_time=$(echo $total_wall_time + $wall_time | bc)
+
+    # tear down
+    cd ..
```
Comment (ekopylova) [inline benchmark/run_phylomizer.sh]: Is this actually doing what you mean by your comment "tear down"? I don't see any file removal..

File: benchmark/run_phylomizer.sh (hunk 1 above)
Comment (qiyunzhu) [reply benchmark/run_phylomizer.sh:40]: No, the default is 3, but apparently, merely 3 taxa are useless for HGT inference. I think we can discuss on this number in the upcoming meeting.

File: benchmark/run_phylomizer.sh (hunk 2 above)
Comment (qiyunzhu) [reply benchmark/run_phylomizer.sh]: There isn't any tearing down here. This is another question I wish that you can comment on: do you clear up program output files after each run? Currently I leave them as is.

**Comment by qiyunzhu** [comment]
@wasade do you have some comments on the protocol too?

File: benchmark/run_phylomizer.sh [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    # only maximum likelihood, but not neighbor joining, is to be conducted
+    echo "
+    verbose             parameter    1
+    residue_datatype    parameter    protein
+    force_seed_sequence parameter	   True
+    alignment           mode         kalign muscle mafft
+    consensus           mode         m_coffee
+    trimming            mode         trimal
+    both_direction      parameter	   True
+    min_seqs            parameter	   10
```
Comment (tanaes) [inline benchmark/run_phylomizer.sh]: Nitpicking, but is there a reason for the strange alignment of the third column here?

File: benchmark/run_phylomizer.sh [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+TIMEFORMAT='%U %R'
+total_user_time="0.0"
+total_wall_time="0.0"
+
+printf "#Phylomizer\n" >> $output_fp
+
+i=0
+cd ${working_dir}/phylomizer
+for gene_fa in $gene_fa_dir/*.fa
```
Comment (tanaes) [inline benchmark/run_phylomizer.sh:88]: Will this not give the full dir path ($gene_fa_dir and up) as $gene_fa? Don't you want to do some manipulation to actually just get the gene and not the path?

```
>>> for bar in $foodir/*.tsv
>>> do
>>>     echo $bar
>>>     echo ${bar%.tsv}
>>>     g=${bar##*/}
>>>     echo ${g%.tsv}
>>> done

./quast/report.tsv
./quast/report
report
./quast/transposed_report.tsv
./quast/transposed_report
transposed_report

```

Comment (qiyunzhu) [reply benchmark/run_phylomizer.sh:88]: Good point! You are definitely right. Let me fix.

File: benchmark/run_phylomizer.sh (hunk 3 above)
Comment (qiyunzhu) [reply benchmark/run_phylomizer.sh]: They are aligned in Atom. Not sure why. Let me realign...

**Review by qiyunzhu — COMMENTED** [review]
@tanaes Thank you for such a careful review!

File: benchmark/run_phylomizer.sh (hunk 3 above)
Comment (qiyunzhu) [inline benchmark/run_phylomizer.sh]:
```
    verbose             parameter    1
    residue_datatype    parameter    protein
    force_seed_sequence parameter    True
    alignment           mode         kalign muscle mafft
    consensus           mode         m_coffee
    trimming            mode         trimal
    both_direction      parameter    True
    min_seqs            parameter    10
    in_letter           parameter    U:B
    in_letter           parameter    O:Z
    muscle              binary
    muscle_params       parameter
    mafft               binary
    mafft_params        parameter    --auto
    kalign              binary
    kalign_params       parameter    -f fasta
    m_coffee            binary       t_coffee
    m_coffee_params     parameter    -n_core 1 -output fasta -quiet
    trimal              binary
    trimal_params       parameter    -phylip -gt 0.1
    trimal_compare      parameter    -ct 0.1667
    readal              binary
    tree                mode         phyml
    evol_models         parameter    JTT WAG MtREV VT LG Blosum62 Dayhoff
    numb_models         parameter    2
    tree_approach       mode         ml
    ml                  parameter    -b -2 -o tlr
    phyml               binary
    phyml_params        parameter    -d aa -f e -v e -a e -c 4 --no_memory_check
```

**Review by tanaes — APPROVED** [review]

**Comment by qiyunzhu** [comment]
@ekopylova Thank you very much!!

**Pull request merged**